crates/cairo-coverage/src/input/mappings/sierra_ids.rs

```diff
@@ -58,3 +59,27 @@ fn compile_to_casm(program: &Program) -> Result<CairoProgram> {
     )
     .context("Failed to compile sierra to casm")
 }
+
+trait Squash {
+    fn squash(self, statement_map: &StatementMap) -> SierraIds;
+}
+
+impl<I> Squash for I
+where
+    I: Iterator<Item = StatementIdx>,
+{
+    fn squash(self, statement_map: &StatementMap) -> SierraIds {
+        SierraIds(
+            self.fold(Vec::new(), |mut acc, statement_idx| {
+                if acc.last().map_or(true, |last_idx| {
+                    statement_map.get(last_idx) != statement_map.get(&statement_idx)
+                }) {
+                    acc.push(statement_idx);
+                }
+                acc
```

Comment:
Probably you want make the lambda a standalone function to give a name to the predicate.